cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup.go

```diff
@@ -95,7 +99,24 @@ func (ng *nodegroup) DeleteNodes(nodes []*apiv1.Node) error {
 		if actualNodeGroup.Id() != ng.Id() {
 			return fmt.Errorf("node %q doesn't belong to node group %q", node.Spec.ProviderID, ng.Id())
 		}
+	}
+
+	// Step 2: if deleting len(nodes) would make the replica count
+	// <= 0, then the request to delete that many nodes is bogus
+	// and we fail fast.
+	replicas, err := ng.scalableResource.Replicas()
+	if err != nil {
+		return err
+	}
+
+	if replicas-len(nodes) <= 0 {
```

Comment:
Would this ever have nodes passed to it whose owning Machine are already marked for deletion, but haven't gone away yet? Should that be taken into account?